src/js/components/ClinPhen/PhenopacketDisplay/BiosampleView.tsx

```diff
@@ -7,46 +7,61 @@ import TDescriptions from '@Util/TDescriptions';
 import type { Biosample } from '@/types/clinPhen/biosample';
 import type { OntologyTerm } from '@/types/ontology';
 
-import { EM_DASH } from '@/constants/common';
 import { useTranslatedTableColumnTitles } from '@/hooks/useTranslatedTableColumnTitles';
+import { hiddenDescriptions } from '@/utils/descriptions';
 
 const BiosampleExpandedRow = ({ biosample }: { biosample: Biosample }) => {
-  const items: DescriptionsProps['items'] = [
+  const items: DescriptionsProps['items'] = hiddenDescriptions([
     {
       key: 'description',
       label: 'biosample_expanded_row.description',
-      children: biosample.description || EM_DASH,
+      children: biosample.description,
     },
     {
       key: 'derived_from_id',
       label: 'biosample_expanded_row.derived_from',
-      children: biosample.derived_from_id || EM_DASH,
+      children: biosample.derived_from_id,
     },
     {
       key: 'sample_type',
       label: 'biosample_expanded_row.sample_type',
       children: <OntologyTermComponent term={biosample.sample_type} />,
+      hidden: !biosample.sample_type,
     },
     {
       key: 'time_of_collection',
       label: 'biosample_expanded_row.collection_time',
-      children: biosample.time_of_collection ? <TimeElementDisplay element={biosample.time_of_collection} /> : EM_DASH,
+      children: <TimeElementDisplay element={biosample.time_of_collection} />,
+      hidden: !biosample.time_of_collection,
     },
     {
       key: 'histological_diagnosis',
       label: 'biosample_expanded_row.histological_diagnosis',
       children: <OntologyTermComponent term={biosample.histological_diagnosis} />,
+      hidden: !biosample.histological_diagnosis,
     },
     {
       key: 'pathological_stage',
       label: 'biosample_expanded_row.pathological_stage',
       children: <OntologyTermComponent term={biosample.pathological_stage} />,
+      hidden: !biosample.pathological_stage,
     },
-  ];
+  ]);
 
   return <TDescriptions bordered size="small" items={items} />;
 };
 
+const isBiosampleRowVisible = (r: Biosample) => {
```

Comment:
clean up by removing {} and return and just using an expression arrow function? we could also make computing items a separate function, but maybe that's too much computational overhead for now.